brushfire-features/src/main/scala/com/stripe/brushfire/features/FeatureEncoder.scala

```diff
@@ -0,0 +1,27 @@
+package com.stripe.brushfire
+package features
+
+// sealed trait FeatureValue
+// object FeatureValue {
+//   case class FString(value: String) extends FeatureValue
+//   case class FDouble(value: Double) extends FeatureValue
+//   sealed abstract class FBoolean(val value: Boolean) extends FeatureValue
+//   case object FTrue extends FBoolean(true)
+//   case object FFalse extends FBoolean(false)
+//   case object FNull extends FeatureValue
+// }
+
+sealed trait FeatureType
+object FeatureType {
+  case object Ordinal extends FeatureType
+  case object Nominal extends FeatureType
+  case object Continuous extends FeatureType
+  case object Sparse extends FeatureType
+}
+
+case class FeatureMapping[A](
+  mapping: Map[String, (FeatureType, A => FeatureValue)]
```

Comment:
`A` here is actually the *record* type, not a value. So `A` is a `Map[String, String]` in the CSV case or some `A <: ThriftStruct` in the thrift case. That said, this could be `A => Map[String, FeatureValue]`. I didn't do this for some reasons initially, but that reason may no longer be valid :D Let me just try to remember my thought process.